### ISSUE TITLE
Add NativeBlobSoftDeleteDeletionDetectionPolicy to Microsoft.Azure.Search library

### ISSUE THREAD

**Issue by xantari** [Azure/azure-sdk-for-net#11435]
**Library or service name.**

Microsoft.Auzre.Search
 
**Is your feature request related to a problem? Please describe.**

It appears that since the 09/2019 release you have had a public preview of "NativeBlobSoftDeleteDeletionDetectionPolicy" delete policy.

However the Microsoft.Azure.Search does not have an ability to define that deletion policy.

Information about Native blob soft delete is here: https://docs.microsoft.com/en-us/azure/search/search-howto-indexing-azure-blob-storage

**Comment by AlexGhiondea** [comment]
/cc @tg-msft for our Track 2 library.

@xantari we have started work on a new version of the Search library -- we are going to consider including this for our upcoming library.

**Comment by heaths** [comment]
Thanks for bringing this to my attention. The new packages do support it, but perhaps not in a very intuitive way. @brjohnstmsft, right now these are exposed as models that just need to exist in the JSON payload. I suggest we make them boolean properties like `Data(Change|Deletion)DetectionPolicyEnabled`. Perhaps we could drop "Policy" to shorten it, but I do like the symmetry with the documentation, such that searching for "DataDeletionDetectionPolicy", for example, yields a match.

**Comment by heaths** [comment]
Sorry, I see those classes should actually be `abstract` (see #9686) and I see that swagger does not actually define @odata.type "#Microsoft.Azure.Search.NativeBlobSoftDeleteDeletionDetectionPolicy". We can define one manually, but this should be in the Preview swagger as well if it's still supported.

**Comment by brjohnstmsft** [comment]
@heaths I defer to @bleroy or @arv100kri for API design opinions in this area.

**Comment by heaths** [comment]
Blocked on Azure/azure-rest-api-specs#9160. This could be done without but requires taking over de/serialization completely instead of using generated code, which continues to evolve. At least, we can wait until time no longer allows for getting a change to swagger.

**Comment by heaths** [comment]
@bleroy, @arv100kri do you have any timeframe for defining this type in swagger?
> Sorry, I see those classes should actually be abstract (see #9686) and I see that swagger does not actually define @odata.type "#Microsoft.Azure.Search.NativeBlobSoftDeleteDeletionDetectionPolicy". We can define one manually, but this should be in the Preview swagger as well if it's still supported.

**Comment by heaths** [comment]
Blocked on the service team, so punting to preview 5 since this is only additive.

**Comment by heaths** [comment]
It doesn't look like this is going to make it, so punting.

**Comment by brjohnstmsft** [comment]
@heaths In cases like this where SDK changes are blocked on Swagger changes, I recommend putting the "Service Attention" tag only on the Swagger issue in azure-rest-api-specs. Otherwise, ownership isn't clear and our issue SLA metrics are affected. I've removed the tag since it's already present on the corresponding azure-rest-api-specs issue.

**Comment by heaths** [comment]
This is tracking https://github.com/Azure/azure-rest-api-specs/issues/9160 which is both in that repo and has the "Service Attention" tag. This is to complete our work once done.

**Comment by jennifermarsman** [comment]
This has been merged into Track 2 SDK.  PR: https://github.com/Azure/azure-rest-api-specs/pull/10308   @heaths can this issue be closed?

**Comment by heaths** [comment]
Yes. Thanks for the reminder. Support for this will be in the Azure.Search.Documents package.

**Comment by psiservices-dbrink** [comment]
@heaths , did this get included in Azure.Search.Documents?  I see no use of the phrase NativeBlobSoftDeleteDeletionDetectionPolicy anywhere in https://github.com/Azure/azure-sdk-for-net.  

Shouldn't it be here with SoftDeleteColumnDeletionDetectionPolicy: 
https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/search/Azure.Search.Documents/src/Generated/Models

**Comment by Mohit-Chakraborty** [comment]
@jennifermarsman, @jadrefke - the change was [removed](https://github.com/Azure/azure-rest-api-specs/commit/e63a68a32b9771014a4083bcabe1939ce1632d1b) a few weeks after it was added and it has been like that since.

Are there plans to add it back?

**Comment by Mohit-Chakraborty** [comment]
We don't have immediate plans to expose this functionality via the SDK. Will update this issue once we add it.
Closing the issue for now.

cc: @giamondragon, @dereklegenzoff

**Comment by xantari** [comment]
Is there a separate issue I should monitor to keep an eye on this since this issue was closed?

**Comment by gmndrg** [comment]
Hello, this issue has been closed since we will reopen when our team is actively working on it. Right now the efforts for this are on hold with no ETA. You can keep following up this thread since we will definitely get back to it when we can allocate the resources to keep working on this accordingly.

You can write anytime here and we'll get back to you gladly with any update available. This is a preview feature. It will have full support when/if going into GA based on https://azure.microsoft.com/en-us/support/legal/preview-supplemental-terms/.

Apologies since we are not able to provide more info at this time. Thank you for understanding.

**Comment by a2741890** [comment]
Is the Azure.Search.Document still being updated? The last beta version released was in November 2021.

**Comment by gmndrg** [comment]
Hello @a2741890, yes, please check https://www.nuget.org/packages/Azure.Search.Documents/#versions-body-tab You can see the latest beta versions and the latest was out 4 days ago.

![image](https://user-images.githubusercontent.com/46359679/178342980-eb37a277-9df8-4304-bb6c-30351339a413.png)

The original thread is already closed. For new threads, please submit a new issue. Thank you.
